lib/galaxy/dependencies/update.sh

```diff
@@ -1,21 +1,55 @@
 #!/bin/sh
-set -e
 
-# You can pass one or more package specifications to this script to add only
-# them without updating all the rest of the requirements.
+# This script creates a temporary venv and installs poetry into that venv.
+# The temporary venv should NOT be activated.
+# It then runs poetry inside Galaxy's root directory, potentially updating 
+# pyproject.toml and poetry.lock, and the pinned and dev requirements files.
+
+set -e
 
 if [ -z "$VIRTUAL_ENV" ]; then
     echo "Please run this script inside a virtual environment!"
     exit 1
 fi
 
-THIS_DIRECTORY="$(cd "$(dirname "$0")" > /dev/null && pwd)"
+this_directory="$(cd "$(dirname "$0")" > /dev/null && pwd)"
+
+usage() {
+    printf "Usage: %s: [-a] [pkg1] [... pkgN]\n" ${0##*/} >&2
+}
+
+add=
+while getopts 'a' OPTION
+do
+    case $OPTION in
+        a) add=1
+           ;;
+        ?) usage
+           exit 2
+           ;;
+    esac
+done
+shift $(($OPTIND - 1))
 
-pip install --upgrade pip setuptools poetry
-if [ $# -gt 0 ]; then
-    poetry add --lock "$@"
+if [ -n "$add" ] && [ $# -eq 0 ]; then
+    printf "When adding (-a), you must provide at least one package name.\n" >&2
+    usage
+    exit 2
+fi
+
+# Create poetry-venv in a tmp directory; install poetry and make shortcut to executable.
+poetry_venv=$(mktemp -d "${TMPDIR:-/tmp}/poetry_venv.XXXXXXXXXX")
+python3 -m venv "${poetry_venv}"
+${poetry_venv}/bin/pip install --upgrade pip poetry
+poetry="${poetry_venv}/bin/poetry"
+
+# Run poetry (this may update pyproject.toml and poetry.lock).
+if [ -z "$add" ]; then
+    "$poetry" update --lock "$@"
```

Comment:
IIRC when I tested poetry, adding a package name to `poetry update` didn't update just that package (i.e. it was equivalent to a simple `poetry update`). Did this work for you?

Reply:
I'm almost certain it did, at least with the lock file present. It still updated multiple packages, but that, I assume, was the consequence of  dependency resolution for the updated package. The docs [state this explicitly](https://python-poetry.org/docs/cli/#update). I don't remember the exact details, but my notes include "update foo does not update all", so it must have worked.

Reply:
OK, good to know! Any reason for dropping the `-v` option? I think it was useful to better understand the reason for downgrades or dep conflicts.

Also, can you please rebase the changes in a single commit so we can get rid of adding and removing the lock file?

Reply:
No reason, just an oversight on my part. Agreed that verbose is better. Adding "-v" does, in fact, add some output, but that extra info is negligible, I think. The "-vv" and "-vvv" options make a difference and ran without issue, but the root cause of the bug has not been removed upstream, so I wouldn't enable those options just yet.
Rebasing.